packages/cfd/src/Containers/jurisdiction-modal/jurisdiction-modal.tsx

```diff
@@ -15,11 +17,51 @@ const JurisdictionModal = observer(({ openPasswordModal }: TJurisdictionModalPro
 
     const { account_type, is_jurisdiction_modal_visible, toggleJurisdictionModal } = useCfdStore();
 
-    const modal_title = show_eu_related_content
-        ? localize('Choose a jurisdiction for your Deriv MT5 CFDs account')
-        : localize('Choose a jurisdiction for your Deriv MT5 {{account_type}} account', {
-              account_type: localize(getMT5Title(account_type.type)),
-          });
+    const [is_dynamic_leverage_visible, setIsDynamicLeverageVisible] = React.useState(false);
+
+    const toggleDynamicLeverage: React.MouseEventHandler<HTMLSpanElement> = event => {
+        event.stopPropagation();
+        setIsDynamicLeverageVisible(!is_dynamic_leverage_visible);
+    };
```

Comment:
I understand that `toggleDynamicLeverage` need to be passed to multiple components. You can use context coupled with a hook if you decide to split the components like below

Reply:
Updated.